### ISSUE TITLE
Jackson Se/De-serliazation question / bug?

### ISSUE THREAD

**Issue by javapapo** [jwtk/jjwt#404]
Hello  I am using the following deps

``` xml
  <jjwt.version>0.10.5</jjwt.version>
<dependency>
      <groupId>io.jsonwebtoken</groupId>
      <artifactId>jjwt-api</artifactId>
      <version>${jjwt.version}</version>
    </dependency>
    <dependency>
      <groupId>io.jsonwebtoken</groupId>
      <artifactId>jjwt-jackson</artifactId>
      <version>${jjwt.version}</version>
      <scope>compile</scope>
    </dependency>
    <dependency>
      <groupId>io.jsonwebtoken</groupId>
      <artifactId>jjwt-impl</artifactId>
      <version>${jjwt.version}</version>
      <scope>runtime</scope>
 </dependency>
```

In the code , I pass my already configured ObjectMapper (resides within SpringBoot init app)

``` java

 //create a POJO and set it on a Map
 Customer obj = new Customer();
 customer.setId(1);  
 Map<String, Object> claims = new HashMap<>();
 claims.put("customerClaim", obj);

// create builder and set the claims map
String token =Jwts.builder().serializeToJsonWith(new JacksonSerializer<>(objectMapper))
                    .signWith(key)
                    .setClaims(claims).compact();
//All GOOD
```
But when I try to retrieve my `Customer` obj I get a class cast exception

``` java

Jws<Claims> jws = Jwts.parser()
                                        .deserializeJsonWith(new JacksonDeserializer<>(objectMapper))
                                        .setSigningKey(key)
                                       .parseClaimsJws(jwtToken);
T payload = jws.getBody().get("customerClaim",Customer.class);
```
But I get

``` java
io.jsonwebtoken.RequiredTypeException: Expected value to be of type: class com.xxxx.Customer, but was class java.util.LinkedHashMap
```

So in the debugger the object in the custom claim is actually a LInkedHashMap.

Do I miss something from the docs?

Maybe its my bad

**Comment by lhazlewood** [comment]
Hi!

This isn't a bug per se - it's more that the feature does not exist just yet.  I think your issue will be addressed by #369 (you might want to read the referenced #214 discussion thread - it shares a lot of insight into what's going on).

As a result, I'm closing this as a duplicate just for housekeeping purposes, but if you feel otherwise, please do make a comment and I'll be happy to open it.  Thanks!

**Comment by javapapo** [comment]
Thanks thats fine I will follow #369 & #214 . Thanks for the prompt response
